grpc-sys/src/lib.rs

```diff
@@ -0,0 +1,343 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![cfg_attr(feature = "dev", feature(plugin))]
+#![cfg_attr(feature = "dev", plugin(clippy))]
+#![cfg_attr(not(feature = "dev"), allow(unknown_lints))]
+
+extern crate libc;
+
+use libc::{c_char, c_int, c_void, int32_t, int64_t, size_t, uint32_t, uint8_t};
+use std::time::Duration;
+
+#[repr(C)]
+pub enum GprClockType {
+    Monotonic = 0,
+    Realtime,
+    Precise,
+    Timespan,
+}
+
+#[repr(C)]
+pub struct GprTimespec {
+    tv_sec: int64_t,
+    tv_nsec: int32_t,
+    clock_type: GprClockType,
+}
+
+impl From<Option<Duration>> for GprTimespec {
```

Comment:
Why not `From<Duration>`? As for `Option<Duration>`, we can `option.map(From::from).unwrap()`.